pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -36,24 +37,41 @@ var (
 )
 
 type Gatherer struct {
-	client     configv1client.ConfigV1Interface
-	coreClient corev1client.CoreV1Interface
+	client        configv1client.ConfigV1Interface
+	coreClient    corev1client.CoreV1Interface
+	metricsClient rest.Interface
 
 	lock        sync.Mutex
 	lastVersion *configv1.ClusterVersion
 }
 
-func New(client configv1client.ConfigV1Interface, coreClient corev1client.CoreV1Interface) *Gatherer {
+func New(client configv1client.ConfigV1Interface, coreClient corev1client.CoreV1Interface, metricsClient rest.Interface) *Gatherer {
 	return &Gatherer{
-		client:     client,
-		coreClient: coreClient,
+		client:        client,
+		coreClient:    coreClient,
+		metricsClient: metricsClient,
 	}
 }
 
 var reInvalidUIDCharacter = regexp.MustCompile(`[^a-z0-9\-]`)
 
 func (i *Gatherer) Gather(ctx context.Context, recorder record.Interface) error {
 	return record.Collect(ctx, recorder,
+		func() ([]record.Record, []error) {
+			if i.metricsClient == nil {
+				return nil, nil
+			}
+			data, err := i.metricsClient.Get().AbsPath("federate").
+				Param("match[]", "ALERTS").
+				Param("match[]", "etcd_object_counts").
```

Comment:
you also promised the cluster_installer? ;-)

Reply:
Yup, need a list.

Reply:
Will do the three for now, then take feedback tomorrow and Friday as you suggested.

Reply:
- [ ] aggregator_unavailable_apiservice_count
- [ ] aggregator_unavailable_apiserver_gauge
- [ ] work queue metrics for kube-controller-manager, kube-scheduler, and kube-apiserver processes (can we do pattern matches?)

@mfojtik we should generally collect CSRs